lib/ansible/module_utils/common/parameters.py

```diff
@@ -838,6 +838,9 @@ def env_fallback(*args, **kwargs):
 
 def set_fallbacks(argument_spec, parameters):
     no_log_values = set()
+    if not isinstance(parameters, Mapping):
+        # The wrong type is handled somewhere else
+        return no_log_values
```

Comment:
This shouldn't be silently ignored. It should be an error at "somewhere else" or here directly.

Reply:
This should raise a `TypeError` if any of the arguments is invalid. But yeah, the input validation must happen in the outer layer, in the calling code somewhere.

Reply:
Actually the caller checked for the right type, but only **after** calling this function. I've switched the order.